### ISSUE TITLE
libdoc parse failed when procedure are having decorators

### ISSUE THREAD

**Issue by vkosuri** [robotframework/robotframework#2286]
I was trying to generate libdoc for [RequestsKeywords.py] (https://github.com/bulkan/robotframework-requests/blob/master/src/RequestsLibrary/RequestsKeywords.py) file, the procedure which are having decorators those all are failed.

``` Shell
python -m robot.libdoc RequestsKeywords.py  RequestsLibrary.html
```

Screen shot

![libdoc](https://cloud.githubusercontent.com/assets/13664257/12085192/50a63376-b2e3-11e5-9a99-c02f2169fc6c.png)

Code snippet 
``` Python
    @retry(RetryException, logger)
    def get_request(self, alias, uri, headers=None, params={}, allow_redirects=None, timeout=None):
        """ Send a GET request on the session object found using the
        given `alias`

        `alias` that will be used to identify the Session object in the cache

        `uri` to send the GET request to

        `headers` a dictionary of headers to use with the request
        
        `timeout` connection timeout
        """        
```

I am using ```Robotframework == 3.0```

**Comment by pekkaklarck** [comment]
The problem is that this decorator doesn't preserve the original signature:
http://robotframework.org/robotframework/latest/RobotFrameworkUserGuide.html#using-decorators

**Comment by Ka55i0peia** [comment]
Referring [this post](https://stackoverflow.com/questions/147816/preserving-signatures-of-decorated-functions) the preservation of signature by `functools.wraps` is available since python 3.4

**Comment by pekkaklarck** [comment]
That's true. #3027 will add this support in RF 3.2.
